Setup changes

clbarnes opened the pull request:
A suite of speculative changes from which you can pick what you like!

- bump python_requires so that it reflects what's tested against
- read the long_description directly from the readme so only one needs to be maintained
- include the optional and test dependencies in the `requirements.txt`, but change `setup.py`'s parsing of it to put them behind `extras` (so `pip install navis` does the same as it always has, but now `pip install navis[igraph]` will install that optional dependency, and the CI uses `pip install -e .[dev]` to install the lint and test dependencies as well). This helps contributors set up a full development environment without having to dig through all the CI files to find extra dependencies, and makes the optional dependencies more transparent for users.
- restrict `find_packages` to the `navis` directory, so that other directories which contain python files (e.g. `tests/`) are not assumed to be packages as well.

schlegelp commented:
I seem to remember I made a "short" long description instead of just reading the `README.md` (which is what I do in all other packages) on purpose. Can't remember why though - so let's go with your changes.

They Python version is set to >=3.6 on purpose because the R-Python interface that some people are using comes with Python 3.6 by default and it's apparently non-trivial to install a more recent version. Will bump in the future when this is less of a problem - 3.6 reaches EOL end of this year after all.

Adding extras is a great idea. Just need to document those somewhere? Perhaps minimally in the readme?

All other changes look good! I'd merge after you change `python_requires` back to `>=3.6`.

clbarnes commented:
Extras are documented and I added extras for the other documented optional deps (rpy2, shapely, pyoctree, pykdtree).

Minor things: changed the doctests so they don't write into the source directory any more (because I am an awful git user and kept `git add .`ing test output...), and allowed str-like formatting of `Path`s in `write_swc`.

schlegelp commented:
Looks good! Thanks Chris!
